### PR TITLE
Implement ELU

### LINKED ISSUE #694: ELU activation function
Djork-Arné Clevert, Thomas Unterthiner and Sepp Hochreiter propose a new activation function which seems to work well:

Fast and Accurate Deep Network Learning by Exponential Linear Units (ELUs)
http://arxiv.org/abs/1511.07289

The ELU (eq. 16) and its gradient (eq. 17) should be very easy to implement.

### PR DESCRIPTION
**Pull request by muupan**
This PR fixes #694.

### CODE REVIEW

File: chainer/functions/activation/elu.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class ELU(function.Function):
+
+    """Exponential Linear Unit."""
```
Comment (delta2323) [inline chainer/functions/activation/elu.py]: Could you add the reference to ELU?

File: chainer/functions/activation/elu.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        gx = cuda.elementwise(
+            'T x, T gy, T alpha', 'T gx',
+            'gx = x >= 0 ? gy : gy * alpha * exp(x)', 'elu_bwd')(
+                x[0], gy[0], self.alpha)
+        return gx,
+
+
+def elu(x, alpha=1.0):
+    """Exponential Linear Unit function.
+
```
Comment (delta2323) [inline chainer/functions/activation/elu.py]: Could you add the reference to ELU?

File: tests/chainer_tests/functions_tests/activation_tests/test_elu.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+class TestELU(unittest.TestCase):
+
+    def setUp(self):
+        # Avoid unstability of numeraical grad
+        self.x = numpy.random.uniform(-1, 1, (5, 4)).astype(numpy.float32)
+        for i in range(self.x.size):
+            if -0.01 < self.x.flat[i] < 0.01:
+                self.x.flat[i] = 0.5
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/activation_tests/test_elu.py]: Unlike ReLU, ELU is differentiable at x = 0. So I do not think we need this avoid sampling around x = 0.

**Comment by delta2323** [comment]
LGTM except the comment.

**Comment by muupan** [comment]
ELU is not differentiable at x = 0 if alpha != 1. Should I add a test case where alpha is exactly 1?

**Comment by delta2323** [comment]
I am sorry. I missed `alpha`. You do not have to treat 1 as a special case.

**Comment by delta2323** [comment]
LGTM

**Pull request merged**